x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -242,26 +242,35 @@ export class CsvGenerator {
   /*
    * Format a Datatable into rows of CSV content
    */
-  private generateRows(
+  private async generateRows(
     columns: string[],
     table: Datatable,
     builder: MaxSizeStringBuilder,
     formatters: Record<string, FieldFormat>,
     settings: CsvExportSettings
   ) {
     this.logger.debug(`Building ${table.rows.length} CSV data rows...`);
+
+    const asyncGenerateRow = async (dataTableRow: Record<string, any>): Promise<string> => {
+      return new Promise((resolve) => {
+        setImmediate(() => {
```

Comment:
Could we add a comment explaining why we are using `setImmediate` here? Perhaps a link to the to nodejs documentation too.

Reply:
https://github.com/elastic/kibana/pull/120309/commits/0d124e22db40a68bcc8b593efbfe21f6a446a4dc